### PR TITLE
Add action to Settings > Advanced > Tools to paste a transaction hash to check transaction state against RPC (not Etherscan) #4003

### LINKED ISSUE #4003: Add action to Settings > Advanced > Tools to paste a transaction hash to check transaction state against RPC (not Etherscan)
Like the Infura ping, but paste a transaction hash into the popup and buzz Infura for transactions state and display it

### PR DESCRIPTION
**Pull request by oa-s**
Closes #4003

### CODE REVIEW

File: AlphaWallet/Settings/Coordinators/CheckTransactionStateCoordinator.swift [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            notificationAlertController = presentationViewController.displaySuccess(title: title, message: message)
+        }
+    }
+}
+
+extension CheckTransactionStateCoordinator: SelectTransactionHashViewControllerDelegate {
+
+    func didSelectedCheckTransactionStatus(in viewController: CheckTransactionStateViewController, transactionHash: String) {
+        guard let server = serverSelection.asServersArray.first else { return }
+        guard let web3 = try? getCachedWeb3(forServer: server, timeout: 6) else { return }
```
Comment (hboon) [inline AlphaWallet/Settings/Coordinators/CheckTransactionStateCoordinator.swift:68]: Can we move more of this logic to a non-UI logic object? Like `AddCustomChain`. Then keep the coordinator more as glue for UIViewControllers (like storyboards are used for navigation), and simpler.

File: AlphaWallet/Transactions/Types/TransactionItemState.swift [hunk 2]
```diff
@@ -12,4 +12,14 @@ enum TransactionState: Int {
     init(int: Int) {
         self = TransactionState(rawValue: int) ?? .unknown
     }
+
+    var description: String {
+        switch self {
+        case .completed: return "Completed"
```
Comment (hboon) [inline AlphaWallet/Transactions/Types/TransactionItemState.swift:18]: Need to add these to localization strings files too?

File: AlphaWallet/Settings/ViewModels/ToolsViewModel.swift [hunk 3]
```diff
@@ -18,13 +19,15 @@ struct ToolsViewModel {
...
-
+    case checkTransactionState
     var title: String {
         switch self {
         case .console:
             return R.string.localizable.aConsoleTitle()
         case .pingInfura:
             return R.string.localizable.settingsPingInfuraTitle()
+        case .checkTransactionState:
+            return "Check Transaction State"
```
Comment (hboon) [inline AlphaWallet/Settings/ViewModels/ToolsViewModel.swift:30]: Need to localize.

File: AlphaWallet/Settings/ViewControllers/ServersViewController.swift [hunk 4]
```diff
@@ -3,11 +3,23 @@
 import UIKit
 
 enum ServerSelection {
+    case unselected
```
Comment (hboon) [inline AlphaWallet/Settings/ViewControllers/ServersViewController.swift:6]: Why do we need to introduce `.unselected`?

**Review by hboon — CHANGES_REQUESTED** [review]

File: AlphaWallet/Settings/ViewControllers/ServersViewController.swift (hunk 4 above)
Comment (oa-s) [reply AlphaWallet/Settings/ViewControllers/ServersViewController.swift:6]: nope, missed to remove it, thanks!

**Review by hboon — APPROVED** [review]
Didn't check. :P

**Pull request merged**